src/shared/install.c

```diff
@@ -1767,6 +1771,10 @@ static int install_context_mark_for_removal(
 
                 r = install_info_traverse(scope, c, paths, i, SEARCH_LOAD|SEARCH_FOLLOW_CONFIG_SYMLINKS, NULL);
                 if (r == -ENOLINK)
+                        /* we encountered a dangling symlink */
+                        continue;
+                else if (r == -ENOENT && i->auxiliary)
+                        /* some unit specified in Also= or similar is missing */
```

Comment:
This looks like it would only change the behaviour of "disable", but to double-check: enabling a unit with non-existing `Also=` should still fail with an error message, and IMHO that is the right thing to do.